km/km_mem.h

```diff
@@ -116,11 +122,27 @@ static const uint64_t GUEST_ARG_MAX = 32 * KM_PAGE_SIZE;
  * address of the first byte in it. Note size equals base in the first half of the space
  *
  * Memory regions that become guest physical memory are allocated using mmap() with specified
- * address, so that contiguous guest physical memory becomes contiguous in KM space as well. We
- * randomly choose to start guest memory allocation from 0x100000000000, which happens to be 16TB.
- * It has an advantage of being numerically the guest physical addresses with bit 0x100000000000 set.
+ * address, so that contiguous guest physical memory becomes contiguous in KM space as well.
+ *
+ * We could randomly choose to start guest memory allocation from something like 0x100000000000,
+ * which happens to be 16TB. It has an advantage of being numerically the guest physical addresses
+ * with bit 0x100000000000 set, so it is easy to visually convert gva and gpa. km could be regular
+ * static, dynamic, or PIE executable. Use -DKM_GPA_AT_16T=1
+ *
+ * On the other hand we could choose this to be 0x0, in which case gpa actually equals to kma. To
+ * avoid collision KM needs to be moved out of the way, of course, which we achieve by making KM a
+ * PIE, static or dynamic.
+ *
+ * Per https://stackoverflow.com/questions/61561331/why-does-linux-favor-0x7f-mappings,
+ * Minimum address for ET_DYN, if we take a look at ELF_ET_DYN_BASE, we see that it is architecture
+ * dependent and on x86-64 it evaluates to:
+ * ((1ULL << 47) - (1 << 12)) / 3 * 2 == 0x555555554aaa
  */
-static const km_kma_t KM_USER_MEM_BASE = (void*)0x100000000000ul;   // 16TiB
+#ifndef KM_GPA_AT_16T
+static const km_kma_t KM_USER_MEM_BASE = (void*)0x0;
```

Comment:
Can we find a more automatic way to propagate changes to this symbol's value to kkm?
Or, can km query kkm to find out if the version that supports the new value is available?